pkg/services/ngalert/api/hcl/hcl.go

```diff
@@ -0,0 +1,30 @@
+package hcl
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+)
+
+type Resource struct {
+	Type string      `hcl:"type,label"`
+	Name string      `hcl:"name,label"`
+	Body interface{} `hcl:",block"`
+}
+
+func Encode(resources ...Resource) (data []byte, err error) {
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
I like this, we use it in the new Alertmanager label matchers parser too.

Reply:
yeah, Encoder is prone to panic :) when the struct has incorrect tags.